arwen/host/execution.go

```diff
@@ -224,42 +274,41 @@ func (host *vmHost) CreateNewContract(input *vmcommon.ContractCreateInput) ([]by
 	gasForDeployment := runtime.GetVMInput().GasProvided
 	err = runtime.CreateWasmerInstance(input.ContractCode, gasForDeployment)
 	if err != nil {
-		runtime.PopInstance()
-		runtime.PopState()
-		arwen.RemoveHostContext(idContext)
+		host.onExitIndirectCreate(idContext)
```

Comment:
Please revert the changes in `CreateNewContract()`. The exiting tasks will be very specific to success vs. error, and I need them explicit here. There will be some refactoring afterwards though (`host.onExitIndirectCreate()` will likely exist, but in a different form).

Reply:
Reverted.